packages/contracts-ops/contracts/scripts/DeployImplementations.s.sol

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity 0.7.6;
+pragma abicoder v2;
+
+/*//////////////////////////////////////////////////////////////
+                                 IMPORTS
+//////////////////////////////////////////////////////////////*/
+// Core
+import {Home} from "@nomad-xyz/contracts-core/contracts/Home.sol";
+import {Replica} from "@nomad-xyz/contracts-core/contracts/Replica.sol";
+import {GovernanceRouter} from "@nomad-xyz/contracts-core/contracts/governance/GovernanceRouter.sol";
+import {IUpdaterManager} from "@nomad-xyz/contracts-core/contracts/interfaces/IUpdaterManager.sol";
+// Bridge
+import {BridgeRouter, EthereumBridgeRouter} from "@nomad-xyz/contracts-bridge/contracts/BridgeRouter.sol";
+import {BridgeToken} from "@nomad-xyz/contracts-bridge/contracts/BridgeToken.sol";
+import {TokenRegistry} from "@nomad-xyz/contracts-bridge/contracts/TokenRegistry.sol";
+// Ops
+import {Config} from "../Config.sol";
+import {JsonWriter} from "../JsonWriter.sol";
+// Utilities
+import {Script} from "forge-std/Script.sol";
+
+abstract contract DeployImplementationsLogic is Script, Config {
+    /*//////////////////////////////////////////////////////////////
+                          DEPLOYED CONTRACTS
+    //////////////////////////////////////////////////////////////*/
+
+    Home home;
+    Replica replica;
+    GovernanceRouter governanceRouter;
+    BridgeRouter bridgeRouter;
+    TokenRegistry tokenRegistry;
+    BridgeToken bridgeToken;
+
+    /*//////////////////////////////////////////////////////////////
+                       IMPLEMENTATION DEPLOYMENT
+    //////////////////////////////////////////////////////////////*/
+
+    // NOTE: these init values do not map to correct expected values.
+    // Storage variables in implementation contracts don't matter.
+    // Purpose is to initialize the implementations as a matter of best practice,
+    // despite the fact that in Nomad's architecture,
+    // un-initialized implementations can't harm the protocol
+    // (unless, in the future, we introduce delegatecall in any implementations)
+    function deployImplementations(string memory _domain) internal {
+        // Home
+        home = new Home(getDomainNumber(_domain));
+        home.initialize(IUpdaterManager(address(getUpdaterManager(_domain))));
+        // Replica
+        replica = new Replica(getDomainNumber(_domain));
+        replica.initialize(0, address(0), bytes32(0), 0);
+        // GovernanceRouter
+        governanceRouter = new GovernanceRouter(
+            getDomainNumber(_domain),
+            getRecoveryTimelock(_domain)
+        );
+        governanceRouter.initialize(
+            address(getXAppConnectionManager(_domain)),
+            address(0)
```

Comment:
is this correct?

Reply:
oh. you're initializing the implementation? but with some real data?

Reply:
the `xAppConnectionManager` is queried, so have to pass a real address there, but the second address (`recoveryManager`) is not queried, so it can be whatever. since it's the implementation, and this is `initialize` / not `constructor`, values don't matter as long as the tx doesn't revert

Reply:
https://github.com/nomad-xyz/monorepo/pull/527/commits/ee6232b67f40972eefbcb7e7900b131aca5ff145